### PR TITLE
Enable document generation using rosdoc2 for ament_python pkgs

### PR DESCRIPTION
**Pull request by Yadunund**
Fix missing `exec_depends` in `package.xmls` and docstrings to generate documentation without any warnings using `autodoc`

### CODE REVIEW

File: examples_tf2_py/package.xml [hunk 1]
```diff
@@ -15,7 +15,11 @@
   <author email="sloretz@openrobotics.org">Shane Loretz</author>
   <author email="tfoote@osrfoundation.org">Tully Foote</author>
 
+  <exec_depend>geometry_msgs</exec_depend>
   <exec_depend>launch_ros</exec_depend>
+  <exec_depend>rclpy</exec_depend>
+  <exec_depend>sensor_msgs</exec_depend>
+  <exec_depend>tf2_ros</exec_depend>
```
Comment (clalancette) [inline examples_tf2_py/package.xml]: Hm, this one shouldn't be necessary; this is strictly a python package, so the `tf2_ros_py` dependency should be enough.  The same goes for a few places below.

**Review by clalancette — CHANGES_REQUESTED** [review]

**Comment by Yadunund** [comment]
Thanks for the review! I've moved `tf2_ros` to `<doc_depend>`. I understand now that the `tf2_ros_py` module is imported as `tf2_ros` for backwards compatibility.

The `sphinx_builder.py` script also needs this patch https://github.com/Yadunund/rosdoc2/commit/226f90562be408979902d5102d68f96ced173ffc to include `<doc_depends>` into the `autodoc_mock_imports` list. I've updated the Checklist in `rosdoc2` to ensure we land this patch.

**Comment by clalancette** [comment]
> I've moved tf2_ros to <doc_depend>

Out of curiousity, why did you choose this route rather than a custom `conf.py`?  I'm fine with either solution, but I do think we should expand the comment a bit to explain that modules `import tf2_ros` while the package is named `tf2_ros_py`.

**Comment by Yadunund** [comment]
To my knowledge even with a custom `conf.py` file, `autodoc` will complain that it cannot import `tf2_ros` when documenting [such modules](https://github.com/ros2/geometry2/blob/cb34d7e25406d66b32b146743448bd9d6638c317/examples_tf2_py/examples_tf2_py/async_waits_for_transform.py#L18-L20). 

In my latest commit, I added custom `rosdoc2.yaml` + sphinx source files to `tf2_ros_py`. We now have documentaion for python `tf2_ros` module 🎉 

Screenshot from 2023-03-14 13-50-27

**Review by clalancette — APPROVED** [review]

**Comment by clalancette** [comment]
CI:
* Linux [Build Status](http://ci.ros2.org/job/ci_linux/18225/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/12756/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/18894/)

**Pull request merged**